python/private/pypi/extension.bzl

```diff
@@ -196,10 +197,16 @@ def _create_whl_repos(module_ctx, pip_attr, whl_map, whl_overrides, group_map, s
         ),
         get_index_urls = get_index_urls,
         python_version = major_minor,
+        evaluate_markers = lambda requirements: evaluate_markers(
```

Comment:
I'm curious: how does this affect things? e.g. the resulting lock file, caching, performance, etc?

When I was trying to get local runtimes integrated with bzlmod, an issue I ran into was needing to know the python version/interpreter path during bzlmod phase, so ended up having to call the interpreter during the bzlmod phase. I couldn't tell if that was a good idea or not, though; the implications weren't very clear to me.

Reply:
From my local testing, it has some overhead, but if we do this once per `pip.parse` invocation, the overhead is not so great. Maybe we could later do this even less, but that requires further code refactoring.